draft-ietf-mls-protocol.md

```diff
@@ -2622,16 +2622,36 @@ struct {
     opaque interim_transcript_hash<0..255>;
     Extension extensions<0..2^32-1>;
     HPKEPublicKey external_pub;
+    uint32 signer_index;
+    opaque signature<0..2^16-1>;
 } PublicGroupState;
-```
+~~~
 
 Note that the `tree_hash` field is used the same way as in the Welcome message.
 The full tree can be included via the `ratchet_tree` extension
 {{ratchet-tree-extension}}.
 
-The information above are not deemed public data in general, but applications
-can choose to make them available to new members in order to allow External
-Commits.
+The signature MUST verify using the public key taken from the credential in the
+leaf node at position `signer_index`.  The signature covers the following
+structure, comprised of fields from the PublicGroupState object:
+
+~~~~~
+struct {
+    opaque group_id<0..255>;
+    uint64 epoch;
+    HPKEPublicKey external_pub;
+} PublicGroupStateTBS;
+~~~~~
```

Comment:
Is there a good reason not to sign everything? Would be simpler and more robust to do so

Reply:
I second this question. Following discussion elsewhere on the mailing list, deniability may be pushed to the application level. If deniability was a factor for not signing everything, then it may not apply anymore.

Reply:
That makes sense. Even if we are still aiming for some form of deniability here, I think that we can sign a little more. For example, the cipher_suite, extensions, should be fine. The opaque_signature should be the main issue with respect to deniability. @raphaelrobert : do you have a strong opinion about not signing the transcript hash?

Reply:
The difference between signing the transcript hash and signing just the public key is that you need knowledge of the key schedule (and how it was derived) to tie the public key to the rest of the group state, while (if you don't encrypt handshake messages), the transcript hash can be computed by any party observing the traffic. At least as far as I understand the transcript hash computation. So in terms of deniability it certainly gets easier for the adversary.

Reply:
Sorry for the late reply. I think the PR really just covers the bare minimum of what needs to be signed. My initial proposal was to cover the whole PublicGroupState struct by the signature, except for the tree_hash field.
Just like @br-hale I think including the transcript hash would make this more robust. That being said, @kkohbrok has a very valid point here: the transcript hash can be computed by a passive network observer without any knowledge of group secrets (this is where we are missing the derived epoch IDs we discarded). My intuition is that if we include the transcript hash we might as well include everything.
As @br-hale pointed out, if we go for the beefier version of the parent hash anyway, seeking any deniability guarantees at this level makes little sense.